### PR TITLE
TSK-1914: introduce IN_REVIEW and READY_FOR_REVIEW Task status

### PR DESCRIPTION
**Pull request by ryzheboka**
https://sonarcloud.io/summary/new_code?id=ryzheboka_taskana&branch=TSK-1914

---
Release Notes:

```
TSK-1914: introduce IN_REVIEW and READY_FOR_REVIEW Task status. Extend claim, cancelClaim and complete accordingly.
```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [x] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [x] PR fulfills the ticket
- [x] Edge cases and unwanted side effects are tested
- [x] Readability

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/task/claim/ClaimTaskAccTest.java [hunk 1]
```diff
@@ -201,9 +330,26 @@ void testForceCompleteClaimedTaskOfAnotherUser() throws Exception {
     assertThat(completedTask.getOwner()).isEqualTo("user-1-2");
   }
 
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_forceCompleteTask_When_InReviewByAnotherUser() throws Exception {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/claim/ClaimTaskAccTest.java:335]:
```suggestion
  void should_ForceCompleteTask_When_InReviewByAnotherUser() throws Exception {
```

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
All in all this looks very nice! :rocket: 
more important: This does not break the existing functionality when users don't want to use the review feature.

General remarks:
* Please remove the diff from the archunit store files, since the architecture tests are currently disabled
* I did not manually verify that all number changes match with your newly created tasks. I trust you on this one.
* Question: Did you mirror all claim / cancel claim tasks? I've seen a mirror on most test, but did not verify that you've mirrored all tests.
* Question: Should we, like SonarCloud suggests, use TestFactories to merge those tests? Lets talk about this. I am personally not sure.
* Question: Why didn't you use the test api for the newly created tests? Do you think it still makes sense to migrate the new tests to the test api? Again, let's talk about this :)

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java [hunk 2]
```diff
@@ -1196,15 +1196,24 @@ private static void claimActionsOnTask(TaskSummaryImpl task, String userId, Inst
     task.setModified(now);
     task.setClaimed(now);
     task.setRead(true);
-    task.setState(TaskState.CLAIMED);
+    if (task.getState().equals(TaskState.READY_FOR_REVIEW)
+        || task.getState().equals(TaskState.IN_REVIEW)) {
```
[the remark this answers is not in the document]
Comment (ryzheboka) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:1200]: Looks good :)

File: lib/taskana-core/src/test/java/acceptance/task/claim/ClaimTaskAccTest.java (hunk 1 above)
Comment (ryzheboka) [reply lib/taskana-core/src/test/java/acceptance/task/claim/ClaimTaskAccTest.java:335]: Ok

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java [hunk 3]
```diff
@@ -1196,15 +1196,24 @@ private static void claimActionsOnTask(TaskSummaryImpl task, String userId, Inst
...
+    }
   }
 
   private static void cancelClaimActionsOnTask(TaskSummaryImpl task, Instant now) {
     task.setOwner(null);
     task.setModified(now);
     task.setClaimed(null);
     task.setRead(true);
-    task.setState(TaskState.READY);
+    if (task.getState().equals(TaskState.IN_REVIEW)) {
```
[the remark this answers is not in the document]
Comment (ryzheboka) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:1212]: Ok

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**